libut/Makefile

```diff
@@ -1,33 +1,41 @@
-OBJS=libut.a
-all: $(OBJS) 
+SHELL=/usr/sh
+VERSION=2.0.1
+MAJOR=2
+SOURCES=$(shell echo src/*.c)
+HEADERS=$(shell echo include/*.h)
+OBJS=$(SOURCES:.c=.o)
+all: libut.a libut.so.$(VERSION)
 INCDIR=./include
+LIBDIR=/usr/lib
 CFLAGS+=-I$(INCDIR)
-CFLAGS+=-Wall -Wextra
+CFLAGS+=-Wall -Wextra -Werror -fPIC
 CFLAGS+=-g
+CFLAGS+=$(OPTFLAGS)
 
-libut.a: libut.o utvector.o utmm.o ringbuf.o
+libut.a: $(OBJS)
 	ar r $@ $^
 
-libut.o: src/libut.c $(INCDIR)/libut.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-utvector.o: src/utvector.c $(INCDIR)/utvector.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-utmm.o: src/utmm.c $(INCDIR)/utmm.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
-
-ringbuf.o: src/ringbuf.c $(INCDIR)/ringbuf.h
-	$(CC) $(CFLAGS) $(CPPFLAGS) -c $<
+libut.so.$(VERSION): $(OBJS)
+	$(CC) $(CFLAGS) -shared -Wl,-soname,libut.so.$(MAJOR) $^ -o $@
 
 .PHONY: clean tests install
 
 clean:
-	rm -f $(OBJS) *.o
+	rm -f *.o *.a *.so*
 	make -C tests clean
 
 tests: libut.a
 	make -C tests 
 
-install: libut.a
-	cp $< /usr/local/lib
+install: libut.a libut.so.$(VERSION)
+	mkdir -p $(DESTDIR)$(LIBDIR)
+	mkdir -p $(DESTDIR)$(LIBDIR)/pkgconfig
+	mkdir -p $(DESTDIR)/usr/include/uthash
+	cp libut.a $(DESTDIR)$(LIBDIR)
+	cp libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)
+	ln -sf $(LIBDIR)/libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)/libut.so.$(MAJOR)
+	ln -sf $(LIBDIR)/libut.so.$(VERSION) $(DESTDIR)$(LIBDIR)/libut.so
+	cp ./include/*.h $(DESTDIR)/usr/include/uthash
```

Comment:
okay. I will change it back. I just did "we do because we can do", and you show me the evidences that we can not do. So I am happy to take the advice